### PR TITLE
Optimize the size for tsbuildinfo

### PR DESCRIPTION
**Pull request by sheetalkamat**
This is simplified version of #43079 and additional optimizations for `tsbuildinfo` size

Changes are:
1. `fileNames` : new entry which is list of filenames `tsbuildinfo` will encounter, index of the file name is used as `fileId` everywhere
2. `fileInfos`: is now a list instead of map. `FileFnfo` is for the file at given `fileId` as index
3. Semantic diagnostics and `affectedFilesPendingEmit` also use `fileId` instead of file name
4. fileIdsList: is a list of `fileIds` that `referenceMap` or `exportedModulesMap` will use. Index of this list will be used as `fileIdsListId` for reference map
5. `referenceMap` and `exportedModulesMap` is converted to tuple array with first entry being `fileId` and second one being `fileIdsListId`
6. Instead of emitting `tsbuildinfo` with 2 space indent, its emitted without spaces
7. Testrunner emits the old format of readable `tsbuildinfo` as a `readable.baseline.txt` so that we can verify correctness easily

### CODE REVIEW

**Comment by sheetalkamat** [comment]
@typescript-bot  pack this

**Comment by sheetalkamat** [comment]
@typescript-bot pack this

**Review by sokra — APPROVED** [review]
Looks good, I noticed a few minor nitpicks for the test runner

**Review by amcasey — COMMENTED** [review]
Can you please summarize the high-level changes?  Pulling out numeric identifiers seems like an easy win - what else changed?

File: tests/baselines/reference/tsc/incremental/initial-build/noEmit-changes-composite.js [hunk 1]
```diff
@@ -148,6 +148,9 @@ function someFunc(arguments) {
...
...53710208-import { indirectClass } from './indirectClass';\nnew indirectClass().classC.prop;","signature":"-4882119183-export {};\r\n","affectsGlobalScope":false},{"version":"-8953710208-import { indirectClass } from './indirectClass';\nnew indirectClass().classC.prop;","signature":"-4882119183-export {};\r\n","affectsGlobalScope":false},{"version":"6714567633-export function writeLog(s: string) {\n}","signature":"8117292349-export declare function writeLog(s: string): void;\r\n","affectsGlobalScope":false},{"version":"-19339541508-function someFunc(arguments: boolean, ...rest: any[]) {\n}","signature":"-4920141752-declare function someFunc(arguments: boolean, ...rest: any[]): void;\r\n","affectsGlobalScope":true}],"options":{"composite":true,"project":"./","configFilePath":"./tsconfig.json"},"fileNamesList":[[2],[1]],"referencedMap":[[3,0],[2,1],[4,0]],"exportedModulesMap":[[2,1]],"semanticDiagnosticsPerFile":[0,1,3,2,4,5,[6,[{"file":"./src/nochangefilewithemitspecificerror.ts","start":18,"length":18,"messageText":"Duplicate identifier 'arguments'. Compiler uses 'arguments' to initialize rest parameters.","category":1,"code":2396,"skippedOn":"noEmit"}]]]},"version":"FakeTSVersion"}
```
[the remark this answers is not in the document]
Comment (sheetalkamat) [reply tests/baselines/reference/tsc/incremental/initial-build/noEmit-changes-composite.js:151]: This is new tsbuild info file emitted on disk. The below is readable baseline (converting file ids to their names retaining old format which was readable)

**Review by amcasey — APPROVED** [review]
My review was somewhat shallow, but the explanation made sense and I didn't notice obvious bugs.

**Pull request merged**